README.txt
Return to Castle Wolfenstein single player GPL source release
=============================================================

This file contains the following sections:

GENERAL NOTES
LICENSE

GENERAL NOTES
=============

Game data and patching:
-----------------------

This source release does not contain any game data, the game data is still
covered by the original EULA and must be obeyed as usual.

You must patch the game to the latest version.

Note that RTCW is available from the Steam store at
http://store.steampowered.com/app/9010/

Linux note: due to the game CD containing only a Windows version of the game,
you must install and update the game using WINE to get the game data.

Compiling on win32:
-------------------

A Visual C++ 2008 project is provided in src\wolf.sln.
The solution file is compatible with the Express release of Visual C++.

You will need to execute src\extractfuncs\extractfuncs.bat to generate src\game\g_save.c

You can test your binaries by replacing WolfSP.exe, qagamex86.dll, cgamex86.dll, uix86.dll at the top of the RTCW install

Compiling on GNU/Linux x86:
---------------------------

Go to the src/unix directory, and run the cons script
(cons is a perl based precursor to scons, this is what we were using at the time)

Run ./cons -h to review build options. Use ./cons -- release to compile in release mode.

If problems occur, consult the internet.

Other platforms, updated source code, security issues:
------------------------------------------------------

If you have obtained this source code several weeks after the time of release
(August 2010), it is likely that you can find modified and improved
versions of the engine in various open source projects across the internet.
Depending what is your interest with the source code, those may be a better
starting point.


LICENSE
=======

See COPYING.txt for the GNU GENERAL PUBLIC LICENSE

ADDITIONAL TERMS:  The Return to Castle Wolfenstein single player GPL Source Code is also subject to certain additional terms. You should have received a copy of these additional terms immediately following the terms and conditions of the GNU GPL which accompanied the RTCW SP Source Code.  If not, please request a copy in writing from id Software at id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville, Maryland 20850 USA.

EXCLUDED CODE:  The code described below and contained in the Return to Castle Wolfenstein single player GPL Source Code release is not part of the Program covered by the GPL and is expressly excluded from its terms.  You are solely responsible for obtaining from the copyright holder a license for such code and complying with the applicable license terms.

IO on .zip files using portions of zlib
---------------------------------------------------------------------------
lines	file(s)
4301	src/qcommon/unzip.c
Copyright (C) 1998 Gilles Vollant
zlib is Copyright (C) 1995-1998 Jean-loup Gailly and Mark Adler

  This software is provided 'as-is', without any express or implied
  warranty.  In no event will the authors be held liable for any damages
  arising from the use of this software.

  Permission is granted to anyone to use this software for any purpose,
  including commercial applications, and to alter it and redistribute it
  freely, subject to the following restrictions:

  1. The origin of this software must not be misrepresented; you must not
     claim that you wrote the original software. If you use this software
     in a product, an acknowledgment in the product documentation would be
     appreciated but is not required.
  2. Altered source versions must be plainly marked as such, and must not be
     misrepresented as being the original software.
  3. This notice may not be removed or altered from any source distribution.

MD4 Message-Digest Algorithm
-----------------------------------------------------------------------------
lines   file(s)
289     src/qcommon/md4.c
Copyright (C) 1991-2, RSA Data Security, Inc. Created 1991. All rights reserved.

License to copy and use this software is granted provided that it is identified
as the <93>RSA Data Security, Inc. MD4 Message-Digest Algorithm<94> in all mater
ial mentioning or referencing this software or this function.
License is also granted to make and use derivative works provided that such work
s are identified as <93>derived from the RSA Data Security, Inc. MD4 Message-Dig
est Algorithm<94> in all material mentioning or referencing the derived work.
RSA Data Security, Inc. makes no representations concerning either the merchanta
bility of this software or the suitability of this software for any particular p
urpose. It is provided <93>as is<94> without express or implied warranty of any
kind.

JPEG library
-----------------------------------------------------------------------------
src/jpeg-6
Copyright (C) 1991-1995, Thomas G. Lane

Permission is hereby granted to use, copy, modify, and distribute this
software (or portions thereof) for any purpose, without fee, subject to these
conditions:
(1) If any part of the source code for this software is distributed, then this
README file must be included, with this copyright and no-warranty notice
unaltered; and any additions, deletions, or changes to the original files
must be clearly indicated in accompanying documentation.
(2) If only executable code is distributed, then the accompanying
documentation must state that "this software is based in part on the work of
the Independent JPEG Group".
(3) Permission for use of this software is granted only if the user accepts
full responsibility for any undesirable consequences; the authors accept
NO LIABILITY for damages of any kind.

These conditions apply to any software derived from or based on the IJG code,
not just to the unmodified library.  If you use our work, you ought to
acknowledge us.

NOTE: unfortunately the README that came with our copy of the library has
been lost, so the one from release 6b is included instead. There are a few
'glue type' modifications to the library to make it easier to use from
the engine, but otherwise the dependency can be easily cleaned up to a
better release of the library.

-----------------------------------------------------------------------------
-----------------------------------------------------------------------------
Third party software

* Autodesk FBX SDK - software development kid distributed along with FBX SDK 
version 20113_1
Static libraries
url - http://usa.autodesk.com/adsk/servlet/pc/item?siteID=123112&id=16127237
NOTE: FBX SDK is only needed for compiling "md4builder" solution
-----------------------------------------------------------------------------
* Autodesk Maya SDK - software development kit distributed along with Autodesk 
Maya located in the same location as maya.exe.
-----------------------------------------------------------------------------

docs/changelog.txt
Changelog

12-20-2016 Dusan Jocic <dusanjocic@msn.com>
	- Removed UI_STARTPREGAMECINEMATIC

12-19-2016 Dusan Jocic <dusanjocic@msn.com>
	- More work on UI part of the multiplayer

12-18-2016 Dusan Jocic <dusanjocic@msn.com>
	- Removed user set cvars on server initialization 
	- Updated binary files
	- Removed CVAR_WOLFINFO
	- Added support for lightmaps
	- Fixed problems with lerp tag index

12-17-2016 Dusan Jocic <dusanjocic@msn.com>
	- Fixed problems with clients and single player game-mode 

12-10-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added multiplayer engine source code
	- Integrated cURL 
	- Updated binary files
	- Fixed engine crash with loading AAS cache (.rcd) files
	- Do not generate debug info in release build

12-06-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added reading ".bullet" files into engine

12-05-2016 Dusan Jocic <dusanjocic@msn.com>
	- First check-in of the Bullet Physics engine implementation.
	- Added "physicslib" static library dependencies to engine and to BSP compiler
	- Added "physics_ridgidbrush" and "physics_ridgidmodel"
	- Added possibility that engine read collision model from brush model
	- Added MDC and MD3 support for BSP compiler
	- Added possibility to load map collision from brush model from BSP compiler
	    NOTE: BSP compiler is compiling additional .cm file
	- Added support for Bullet Physics into BSP compiler
		NOTE: BSP compiler is compiling additional .bullet file 
	- Fixed problems with AI 
	- Added VS solution files for Tremulous gamelogics
	- Cleaned library files
	- Update binary files
		NOTE: From this revision loading of maps without recompiling and adding additional .cm and .bullet files will not work
	- Added experimental OpenCL support for BSP compiler
	- Added libCURL library files

12-01-2016 Dusan Jocic <dusanjocic@msn.com>
	- Removed VM
	- Integrated UI into engine

11-28-2016 Dusan Jocic <dusanjocic@msn.com>
	- Moved CM and Sound files in their own folders
	- Removed FBX library and include files
	- Properly linked FBX (fbx20113_1_fbxsdk_win.exe) build inside installed folder on Windows platform
	- Implemented new "MDV" model format with possibility to export models what are created from Maya application
	- Added missing VS solution file
	- Added missing source code for MDVBuilder
	- Added Q3Map2 into source tree
		NOTE: Removed all the extra dependencies and extra stuff from q3map2 including all the XML code, jpeg code and png code.
	- Added Bullet Physics version 2.81 into library folder with compiled library files for Visual Studio 2015

q3map2 is messy fixme...

11-27-2016 Dusan Jocic <dusanjocic@msn.com>
	- Fixed game saving system

11-26-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added initial integration of GLSL backend for OpenGL renderer
		NOTE: Supported are "vertex", "fragment" and "geometry" shaders
			  Engine and renderer is expecting files from the "renderProgs" folder with ".vertex", ".fragment" and ".geometry" extensions.
	- Misc fixes
	- Updated binary files
	- Changed GPUWorkers extension to ".opencl"
	- Added possibility to read a block of memory from the graphics card in OpenCL
	- Updated binary files
	- Renamed all tr_* files into r_*
	- Added updating tangent matrix if it is needed
	- Added support for multiple texture units.
	- Fix crash from reading past end of tr.refdef.drawSurfs
	- Roll up Bugfixes
	- R_LoadCacheImages takes 4 parameters
	- Misc fixes

11-25-2016 Dusan Jocic <dusanjocic@msn.com>
	- Removed bg_lib.c
	- Added OpenCL library files
	- Added "GPUWorkers" for renderer engine
		NOTE: Idea was simple. Adding possibility to use OpenCL along OpenGL for calculating algorithms, ... etc.
		      Engine and renderer is expecting files from the "renderWorkerProgs" folder with ".renderworkerprog" extension.
			  If GPU device is not "somehow" supporting OpenCL, engine will choose CPU as default device.
	- Updated binary files

11-24-2016 Dusan Jocic <dusanjocic@msn.com>
	- Ported whole engine, gamelogic and tools to C++
	- Reformatted code with AStyle to match global indenting and formatting

11-22-2016 Dusan Jocic <dusanjocic@msn.com>
	- Fixed extractfuncs.bat
	- Added support for "WP_MEDIC_SYRINGE"
	- register WolfMP models and skins
	- Increased POOLSIZE for G_Alloc so maps can load
	- Updated binary files
	- Load all .PK3 files inside "main" folder

11-21-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added support for "WP_AMMO" and "WP_PLIERS"
	- Updated binary files
	- Added support for "WP_SMOKE_GRENADE" and some support for multiplayer inside UI
	- Added support for "WP_MEDKIT"
	- Updated binary files

11-20-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added missing script multiplayer actions (G_ScriptAction_ObjectiveShortAxisDesc, G_ScriptAction_ObjectiveShortAlliedDesc, G_ScriptAction_SetDefendingTeam, G_ScriptAction_OverviewImage and G_ScriptAction_RemoveEntity)
	- Added support for "Binoculars"

11-13-2016 Dusan Jocic <dusanjocic@msn.com>
	- Fixed compiling on Windows platform
	- Moved AStyle files in root folder
	- Fixed compiling "tools"
	- Reformatted Java code with AStyle to match global indenting and formatting
	- Added missing Android files
	- Added missing GPL licence, COPYING and README files
	- Removed added folder
	- Added missing Android project layout files
	- Fixed debug build on Windows
	- Fixed crash on engine exit
	- Updated binary files
	- Write info only in debug build

11-12-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added support for converted FBX animated models inside engine and gamelogics
	- Added FBX exporter with possibility to export FBX fiels to ".md4anim" and to ".md4mesh"
	- Added FBX library files
	- Moved "bspc" and "extrafuncs" to "tools" folder
	- Moved game logic files to new folder
	- Renamed engine "src" folder

11-09-2016 Dusan Jocic <dusanjocic@msn.com>
	- Added "Tools" solution folder inside Visual Studio configuration
	- Fixed compiling BSPC project
	- Reformatted code with AStyle to match global indenting and formatting


src/botai/inv.h
//////////////////////////////////////////////////////////////////////////////////////
//
//  This file is part of OWEngine source code.
//  Copyright (C) 1999-2010 id Software LLC, a ZeniMax Media company.
//  Copyright (C) 2016 Dusan Jocic <dusanjocic@msn.com>
//
//  This file is part of the OWEngine single player GPL Source Code.
//
//  OWengine Source Code is free software: you can redistribute it and/or modify
//  it under the terms of the GNU General Public License as published by
//  the Free Software Foundation, either version 3 of the License, or
//  (at your option) any later version.
//
//  OWEngine Source Code is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
//  GNU General Public License for more details.
//
//  You should have received a copy of the GNU General Public License
//  along with OWEngine Source Code. If not, see <http://www.gnu.org/licenses/>.
//
//  In addition, the OWEngine Source Code is also subject to certain additional terms.
//  You should have received a copy of these additional terms immediately following
//  the terms and conditions of the GNU General Public License which accompanied the
//  OWEngine Source Code. If not, please request a copy in writing from id Software
//  at the address below.
//
//  If you have questions concerning this license or the applicable additional terms,
//  you may contact in writing id Software LLC, c/o ZeniMax Media Inc.,
//  Suite 120, Rockville, Maryland 20850 USA.
//
// -------------------------------------------------------------------------
//  File name:   inv.h
//  Version:     v1.00
//  Created:
//  Compilers:
//  Description: Quake3 bot AI
// -------------------------------------------------------------------------
//  History:
//
//////////////////////////////////////////////////////////////////////////////////////

#define INVENTORY_NONE              0
//armor
#define INVENTORY_ARMOR             1
//weapons
#define INVENTORY_LUGER             4
#define INVENTORY_MAUSER            5
#define INVENTORY_MP40              6
#define INVENTORY_SP5               7
#define INVENTORY_ROCKETLAUNCHER    8
#define INVENTORY_GRENADELAUNCHER   9
#define INVENTORY_VENOM             10
#define INVENTORY_FLAMETHROWER      11
#define INVENTORY_CROSS             12

#define INVENTORY_GAUNTLET          13


// please leave these open up to 27 (INVENTORY_9MM) (and double check defines when merging)
// the inventory max (MAX_ITEMS) is 256, so we aren't too concerned about running out of space

//ammo
#define INVENTORY_9MM               27
#define INVENTORY_792MM             28
#define INVENTORY_SP5AMMO           29
#define INVENTORY_ROCKETS           30
#define INVENTORY_GRENADES          31
#define INVENTORY_127MM             32
#define INVENTORY_FUEL              33
#define INVENTORY_CHARGES           34

// please leave these open up to 48 (INVENTORY_HEALTH) (and double check defines when merging)
// the inventory max (MAX_ITEMS) is 256, so we aren't too concerned about running out of space

//powerups
#define INVENTORY_HEALTH            48
#define INVENTORY_TELEPORTER        49
#define INVENTORY_MEDKIT            50
#define INVENTORY_QUAD              51
#define INVENTORY_ENVIRONMENTSUIT   52
#define INVENTORY_HASTE             53
#define INVENTORY_INVISIBILITY      54
#define INVENTORY_REGEN             55
#define INVENTORY_FLIGHT            56
#define INVENTORY_REDFLAG           57
#define INVENTORY_BLUEFLAG          58
//enemy stuff
#define ENEMY_HORIZONTAL_DIST       200
#define ENEMY_HEIGHT                201
#define NUM_VISIBLE_ENEMIES         202
#define NUM_VISIBLE_TEAMMATES       203

//item numbers (make sure they are in sync with bg_itemlist in bg_misc.c)
#define MODELINDEX_ARMORSHARD       1
#define MODELINDEX_ARMORCOMBAT      2
#define MODELINDEX_ARMORBODY        3
#define MODELINDEX_HEALTHSMALL      4
#define MODELINDEX_HEALTH           5
#define MODELINDEX_HEALTHLARGE      6
#define MODELINDEX_HEALTHMEGA       7

#define MODELINDEX_GAUNTLET         8
#define MODELINDEX_SHOTGUN          9
#define MODELINDEX_MACHINEGUN       10
#define MODELINDEX_GRENADELAUNCHER  11
#define MODELINDEX_ROCKETLAUNCHER   12
#define MODELINDEX_LIGHTNING        13
#define MODELINDEX_RAILGUN          14
#define MODELINDEX_SP5              15
#define MODELINDEX_BFG10K           16
#define MODELINDEX_GRAPPLINGHOOK    17

#define MODELINDEX_SHELLS           18
#define MODELINDEX_BULLETS          19
#define MODELINDEX_GRENADES         20
#define MODELINDEX_CELLS            21
#define MODELINDEX_LIGHTNINGAMMO    22
#define MODELINDEX_ROCKETS          23
#define MODELINDEX_SLUGS            24
#define MODELINDEX_BFGAMMO          25

#define MODELINDEX_TELEPORTER       26
#define MODELINDEX_MEDKIT           27
#define MODELINDEX_QUAD             28
#define MODELINDEX_ENVIRONMENTSUIT  29
#define MODELINDEX_HASTE            30
#define MODELINDEX_INVISIBILITY     31
#define MODELINDEX_REGEN            32
#define MODELINDEX_FLIGHT           33
#define MODELINDEX_REDFLAG          34
#define MODELINDEX_BLUEFLAG         35


